packages/core/src/pattern/pattern-config.mjs

```diff
@@ -101,12 +98,14 @@ PatternConfig.prototype.addPart = function (part) {
 
 /** Log the final report on part inheritance order */
 PatternConfig.prototype.logPartDistances = function () {
+  const priorities = []
   for (const partName in this.parts) {
-    let qualifier = DISTANCE_DEBUG ? 'final' : ''
-    this.store.log.debug(
-      `⚪️  \`${partName}\` ${qualifier} options priority is __${this.__mutated.partDistance[partName]}__`
-    )
+    const p = this.__mutated.partDistance[partName]
+    if (p in priorities) priorities[p] += ', ' + partName
+    else priorities[p] = partName
   }
+  for (let p = 1; p < priorities.length; p++)
+    this.store.log.debug(`⚪️  Options priority __${p}__ :\`${priorities[p]}\``)
```

Comment:
Since this is a core change, I'll be a lot more strict 😉 

There's a bunch of bugs that this code invites that are mainly due to the use of an array on line 101.
If the array is sparse, it won't work as expected. Also using `in` on an array seems like a bad idea.

Allow me to propose an alternative:

```mjs
// Use an object here
const priorities = {}
// Using for ... in on an object is frowned upon by purists ;)
for (const partName of (Object.keys(this.parts)) {
  const p = this.__mutated.partDistance[partName]
  // Make sure the property exists in the object, and use a Set to avoid duplicates
  if (typeof priorities[p] === 'undefined') priorities[p] = new Set()
  priorities[p].add(partName)
}

// now iterate over the priorities
for (const p of Object.keys(priorities)) {
  /* 
    * Log the list, we'll turn the set into an array
    * then map() it to add the backticks around the part name 
    * (giving it that code markup) 
    * and finally join() them with a comma and space
    */
  this.store.log.debug(`⚪️ Options priority __${p}__ : ${[...priorities[p]].map(p => '`'+p+'`').join(', ')}`)
}
```

That seems a lot more robust to me.
Seems, because I did not test it.